src/Explore/DocumentList.js

```diff
@@ -20,21 +18,23 @@ const QUERY_CONFIG = {
 
 function DocumentList() {
   const loadOnScroll = useAppStore((state) => state.loadOnScroll)
-  const [query, setQuery] = useAppStore((state) => [
-    state.query,
-    state.setQuery,
-  ])
```

Comment:
Returning a new array in the selector defeats the purpose of having the selector (i.e. zustand will see a new object reference and think that the part of the state retrieved by the hook has changed, and that the component has to be re-rendered, regardless of whether `query` and `setQuery` have changed).

Reply:
You could pass zustand's `shallow` comparator as second parameter to the hook, but I find it easier to just duplicate the hook.